libtransmission/announcer.cc

```diff
@@ -1309,30 +1311,20 @@ static void tierAnnounce(tr_announcer* announcer, tr_tier* tier)
 ****
 ***/
 
-static bool multiscrape_too_big(char const* errmsg)
+static bool multiscrape_too_big(std::string_view errmsg)
 {
     /* Found a tracker that returns some bespoke string for this case?
        Add your patch here and open a PR */
-    static char const* const too_long_errors[] = {
+    auto constexpr TooLongErrors = std::array<std::string_view, 3>{
         "Bad Request",
         "GET string too long",
         "Request-URI Too Long",
     };
 
-    if (errmsg == nullptr)
-    {
-        return false;
-    }
-
-    for (size_t i = 0; i < TR_N_ELEMENTS(too_long_errors); ++i)
-    {
-        if (strstr(errmsg, too_long_errors[i]) != nullptr)
-        {
-            return true;
-        }
-    }
-
-    return false;
+    return std::any_of(
+        std::begin(TooLongErrors),
+        std::end(TooLongErrors),
+        [&errmsg](auto const& toolong) { return errmsg.find(toolong) != std::string_view::npos; });
```

Comment:
Yes, it is. Good catch. :+1:

Right after your review I pushed another commit to use a range-loop here so that I could make the function constexpr, and unfortunately `std::any_of` isn't constexpr in C++17.

If I was keeping the lambda here I would use the trivial copy for std::string_view as the parameter.

Reply:
No real idea what's passed into the function. If it's a value in TooLongErrors, could just do std::find() != std::end().

Although there also is this lovely header only library that adds a .find() method to avoid manual loops: https://github.com/serge-sans-paille/frozen